### PR TITLE
Added support for text chars to parser: %, $, ^, °, >, > and \n.

### PR DESCRIPTION
**Pull request by edrush**
Hey,

another attempt :)

This works well for our application. I don't know about other use cases, which I think you do better...

Regards!

### CODE REVIEW

**Comment by bobvandevijver** [comment]
Hi! This looks promising, I will try to test this tomorrow evening. 

Maybe we should consider adding the package `gensymb` for the degree symbol, what's your opinion on that?

**Comment by bobvandevijver** [comment]
@edrush I did not manage to look at it today, but I will do that later this week! Sorry!

**Comment by edrush** [comment]
@bobvandevijver No prob! For the `gensymb` package: I can not tell what that would imply; I'm not so much in Latex...

**Comment by bobvandevijver** [comment]
@edrush / @jackd248 I see a lot of extra commits, are they supposed to be added to this PR?

File: Helper/Parser.php [hunk 1]
```diff
@@ -36,6 +36,10 @@ public function parseText($text, $checkTable = true) {
     foreach ($matches[0] as $match) {
       $text = str_replace($match, $this->htmlCodes[$match], $text);
     }
+    $text = str_replace("\\", "\\backslash ", $text);
+    $text = str_replace("{", "\\{ ", $text);
+    $text = str_replace("}", "\\} ", $text);
```
Comment (bobvandevijver) [inline Helper/Parser.php]: By merging these (\,{,}) changes, we drop the support for native latex-commands, which is something I do not want to do. We should probably add an option to enable stripping the latex-commands (which enables these lines) in the generator.

File: Helper/Parser.php [hunk 2]
```diff
@@ -104,6 +108,16 @@ public function parseText($text, $checkTable = true) {
 
     $text = str_replace("#", "\\#", $text);
     $text = str_replace("_", "\\_", $text);
+    $text = str_replace("%", "\\%", $text);
+    $text = str_replace("$", "\\$", $text);
```
Comment (bobvandevijver) [inline Helper/Parser.php]: Same holds for the $. I do not mind the #, but those should also be included in the option.

**Comment by bobvandevijver** [comment]
@edrush / @jackd248 I've created an PR (#22), could that work for you?

If so, I will merge that and add the other changes from this PR to the repo.

**Comment by edrush** [comment]
Hey @bobvandevijver, sorry for beeing soooo late: yes, @jackd248's and my commits appeared from the same project. We supposed that the parseText()-function will receive pure text only, and so native latex commands do not matter there. That is why e.g. a backslash must be converted to the corresponding latex command for a backslash. An, maybe useless, example would be that a documentation of a latex usage within a text is to be converted into a latex document. Were we wrong?

**Comment by bobvandevijver** [comment]
@edrush No problem, I was late as well

By default, my users use the WYSIWYG-HTML-editor which can handle most styling: however for example tables are not supported although they are wanted. If we do not allow any LaTeX-commands in the text, it would no longer be possible to create specific constructions that advanced used might want to use. 

However, I also understand the use case in which you purely want to support plain text. So, maybe we should consider creating another 'parse' function that will escape all latex commands. I made an attempt in #22 by adding an argument to escape all latex commands if it's set to true. I can also imagine another Twig filter that will do it automatically. 

There is one thing I won't do: BC-breaking changes. That's why I will not merge this directly, but I am open in creating a solution that fits everyone use case :)

**Comment by edrush** [comment]
Sounds great!

**Comment by bobvandevijver** [comment]
Well, some time has passed (finished my studies and started working, so my time was very limited), but I guess everything is now solved with #22. Will also create a new release in a moment (v1.2.0).

You can create a new issue if you're still missing something, or if something went wrong ;)

**Pull request closed without merging**